cupy/sparse/linalg/solve.py

```diff
@@ -0,0 +1,68 @@
+import numpy
+
+import cupy
+from cupy import cuda
+from cupy.cuda import device
+from cupy.linalg import util
+from cupy.sparse import csr_matrix
+from cupy.sparse import isspmatrix_csr
+
+
+if cuda.cusolver_enabled:
+    from cupy.cuda import cusolver
+
+
+def lsqr(A, b):
+    """Solve linear system with QR decomposition.
+
+    Find the solution to a large, sparse, linear system of equations.
+    Given two-dimensional matrix `A` is decomposed into ``Q * R``.
+
+    Args:
+        A (cupy.ndarray or cupy.sparse.csr_matrix): The input matrix with
+            dimension ``(N, N)``
+        b (cupy.ndarray): Right-hand side vector.
+
+    Returns:
+        x (cupy.ndarray): The final solution.
+
+    .. seealso:: :func:`scipy.sparse.linalg.lsqr`
+    """
+
+    if not cuda.cusolver_enabled:
+        raise RuntimeError('Current cupy only supports cusolver in CUDA 8.0')
+
+    if not isspmatrix_csr(A):
+        A = csr_matrix(A)
+    util._assert_nd_squareness(A)
+    util._assert_cupy_array(b)
+    m, _ = A.shape
```

Comment:
```m = len(A)```

Reply:
`len` can't be called with the sparse matrix `A`. So I changed it to `m = A.shape[0]`.